Docs/StyleGuide/Style-Guide.md

```diff
@@ -0,0 +1,280 @@
+This style guide provides coding conventions for the Dafny code.
+
+*This documentation is still in progress. Please feel free to add any more suggestions if any.*
+
+## Naming Convention
+Any **variables** are named with `camelCase`.
+```dafny
+var minValue := 1;
+var cipherMessage := "Hello World";
+```
+
+Any **lemmas**, **predicates**, **functions**, **methods**, **classes**, and **modules** are named with `PascalCase`.
+```dafny
+method FindIndex(arr: seq<int>, k: int)
+    ...
+```
+
+Any static or global **constants** are named with `UPPERCASE_WITH_UNDERSCORES`.
+```dafny
+static const MONTHS_IN_A_YEAR := 12;
+```
+
+### Method Prefix
+Avoid redundant names when variables or methods are in a class/module.
+```dafny
+class Integer {
+
+    // The following method converts the given integer
+    // to a string.
+    //
+    // this method name can be simplified to ToString()
+    // so that the method call is Integer.ToString(i)
+    // instead of Integer.IntegerToString(i).
+
+    // YES
+    method ToString(i: int) returns (s: string)
+        ...
+
+    // NO
+    method IntegerToString(i: int) returns (s: string)
+        ...
+}
+```
+
+## Code Layout
+
+### Braces
+Dafny by default opens a new brace in the same line.
+```dafny
+module M() {
+    ...
+    method Met() {
+        ...
+    }
+}
+```
+In case the method (or function, lemma, etc) signature is too long to fit in one line, or in case the signature has at
+least one specification clause, the new brace goes to the new line.
+
+```dafny
+module M() {
+    ...
+    method Met(i: int) returns (j: int)
+        requires i % 2 == 0
+        ensures j > 10
+    {
+        ...
+    }
+}
+```
+
+This applies to every scope: `module`, `class`, `predicate`, `if`, `while`, and more.
+
+### Imports
+Import the modules with `opened` to avoid typing a full module name each time a method is invoked.
+```dafny
+import opened StandardLibrary
+...
+```
+
+When there are more than two modules with the same method name, specify the name of the module during method invocation.
+
+```dafny
+import opened MyModule
+import opened YourModule
+...
+method MyMethod() {
+    MyModule.foo();
+    YourModule.foo();
+}
+```
+
+In case typing the full module name is cumbersome, import it with a shorthanded name.
+
+```dafny
+import M = MyModuleWithACumbersomeName
+import Y = YourModuleWithACumbersomeName
+...
+method MyMethod() {
+    M.foo();
+    Y.foo();
+}
+```
```

Comment:
I'd suggest the following for imports:

### Imports

By default, import modules without opening them.

```
import Coffee
...
```

However, if some members of a module are used very frequently, import it using `opened`:

```
import opened Donut
...
```

When a file uses two modules, and both of them define a method of the same name, do not import them `opened`.

```dafny
import MyModule
import YourModule
...
method MyMethod() {
    MyModule.foo();
    YourModule.foo();
}
```

In this case, if you want to shorten their name, import them with a shorthand name.

```dafny
import M = MyModuleWithACumbersomeName
import Y = YourModuleWithACumbersomeName
...
method MyMethod() {
    M.foo();
    Y.foo();
}
```